akka-http-jwt/src/main/scala/akka/http/scaladsl/server/directives/JwtDirectives.scala

```diff
@@ -0,0 +1,61 @@
+package akka.http.scaladsl.server.directives
+
+import akka.event.LoggingAdapter
+import akka.http.jwt.impl.settings.JwtSupport
+import akka.http.jwt.scaladsl
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.server.Directives.Authenticator
+import akka.http.scaladsl.server.Directives.authenticateOAuth2
+import spray.json.{ JsBoolean, JsNumber, JsObject, JsString }
+
+trait JwtDirectives {
+
+  import BasicDirectives._
+  import RouteDirectives._
+
+  def jwt(): Directive1[JwtClaims] = {
+    extractActorSystem.flatMap { system =>
+      jwt(scaladsl.JwtSettings(system))
+    }
+  }
+
+  def jwt(settings: scaladsl.JwtSettings): Directive1[JwtClaims] = {
+    extractLog.flatMap { log =>
+      authenticateOAuth2(settings.realm, bearerTokenAuthenticator(settings.jwtSupport, log)).flatMap { claims =>
+        provide(new JwtClaims(claims))
+      }
+    }
+  }
+
+  private def bearerTokenAuthenticator(jwtSupport: JwtSupport, log: LoggingAdapter): Authenticator[JsObject] = {
+    case p @ Credentials.Provided(token) =>
+      jwtSupport.validate(token) match {
+        case Right(claims) => Some(claims)
+        case Left(ex) =>
+          log.debug("The token was rejected: {}", ex.getMessage)
+          None // FIXME: should we propagate anything else further?
+      }
+  }
+
+  // JwtClaims provides utilities to easily assert and extract claims from the JWT token
+  class JwtClaims(claims: JsObject) {
```

Comment:
This type is now tied to the JwtDirectives trait instance, (path dependent type), which is maybe a bit strange even if it probably doesn't surface as a problem, would probably be fine to put it directly in the directives package, or in the companion object.

Reply:
moved to independent class directives package